corehq/apps/locations/models.py

```diff
@@ -422,6 +422,8 @@ def delete(self, *args, **kwargs):
         for loc in to_delete:
             publish_location_saved(loc.domain, loc.location_id, is_deletion=True)
 
+        delete_locations_filter_criteria.delay([loc.location_id for loc in to_delete])
```

Comment:
Since this is a SQL model, you can also set it up to use a cascading delete like here:
https://github.com/dimagi/commcare-hq/blob/bee1fd160139f21ef3a629268a0d44f3a0e02afa/corehq/apps/app_manager/models.py#L6047-L6047
That way PostgreSQL will handle it for us and we don't need to worry about celery tasks.

Reply:
@esoergel I updated the code, but decided to take another approach after I have rethinked it a bit. This time I remove the rule completely which depends on the location, since the location-dependant rule does not make sense after a location is deleted. If I only remove the definition from the rule, the rule might affect a wrong set of cases since the criteria changed.
As such, I renamed the function and kept it as a celery task to be executed on location delete.